### ISSUE TITLE
Giveaway doesn't send the end message or log in end event!

### ISSUE THREAD

**Issue by oMvhmoud** [Androz2091/discord-giveaways#465]
**Describe the bug**  

The giveaway when ends, doesn't send the end message until I restart the bot.

**Error**  
None.

**To Reproduce**
Create a giveaway and wait until it ends.

**Expected behavior**  
Bot sends the end message and log it in the end event.

**Screenshots**  
https://prnt.sc/mfUQhJWPrC0o

**Additional context**

* `discord-giveaways` version: 5.2.1
* `discord.js` version: 13.8.0
* Custom Database (if you use one): Mongoose

**Comment by oMvhmoud** [comment]
tried quickmongo and the same issue

**Comment by Nico105** [comment]
If you look into your db when the giveaway 1st ended, is `ended = true`?

**Comment by oMvhmoud** [comment]
> And you waited some time before you restarted right?

yes i waited for about 10 mins which is enough, about the ended it's false always until i restart
![image](https://user-images.githubusercontent.com/40049200/172621199-6952ba67-e194-4a3d-b5f3-82768e2fd141.png)

**Comment by Nico105** [comment]
Could you provide your client options?

**Comment by oMvhmoud** [comment]
> Could you provide your client options?

`client.giveawaysManager = new GiveawaysManagerMongoose(client, {
	default: {
		botsCanWin: false,
		embedColor: "#00FFC1",
		reaction: "🎉",
		embedColorEnd: "#ff0000",
		lastChance: {
			enabled: true,
			content: "⚠️ **LAST CHANCE TO ENTER !** ⚠️",
			threshold: 10000,
			embedColor: "#FF0000"
		}
	}
});`

if you mean fucntions it's the same as the mongoose example you provides.

**Comment by Nico105** [comment]
i meant your options for the discord.js client, or you maybe named it "bot"

**Comment by oMvhmoud** [comment]
> i meant your options for the discord.js client, or you maybe named it "bot"

oh, sorry i didn't see that, however im using ur example file [mongoose.js](https://github.com/Androz2091/discord-giveaways/blob/master/examples/custom-databases/mongoose.js) so the client is the same.

**Comment by Nico105** [comment]
does it happen for every giveaway you start?

**Comment by oMvhmoud** [comment]
> does it happen for every giveaway you start?

ofc, also sorry for being reply late, but my pc was dead however hope you help me.

**Comment by Lebyy** [comment]
Can re-produce this the giveaway doesn't end neither fires the giveawayEnded event.
![image](https://user-images.githubusercontent.com/45114019/177693797-7f4ab43a-44cb-43e2-a35c-282af526d6a2.png)

**Comment by Nico105** [comment]
I reverted the changes that are at fault for this. If you update everything should work again. LMK
The slightly weird/annoying thing is, I wasn't able to repro this in my test bot, probably because of its simplicity/low load level.
The thing is, those little persistent problems somehow never affect all users and only some, even when the settings are simple or the same as mine.

Those changes were an attempt to fix issues like #392 #445 and the ones that got closed.
Unfortunately this will leave those open. (I don't even know if those changes would have fixed them since I didn't get to ask yet, but logically the system should have improved the situation).
The issue this time lies in unknown errors/behavior of discord.js message edits, which affected the checking system that now got reverted.
😩
